### PR TITLE
fix most doctest errors

### PR DESCRIPTION
**Pull request by rfourquet**
Most of the errors came from `InexactError/DomainError` which now take arguments, and from `macroexpand` which must be given a module. Sometimes, I had to disable the doctest (using `julia-repl` instead), because at the repl some identifier are prefixed with `Main.`, which is not the case when doctesting. But it could be better to put back `jldoctest` even if it errors out when doctesting, to catch future other errors. For one example (`randn`), the problem came from using too much horizontal space (some values were replaced by horizontal dots); I don't know if the screen width can be parameterized in doctests, but it was simpler to change the array dimensions.

### CODE REVIEW

**Comment by KristofferC** [comment]
xref https://github.com/JuliaLang/julia/pull/22568

**Comment by rfourquet** [comment]
oups, should have checked first... what is the status of this PR?

**Comment by KristofferC** [comment]
Let's just go with this one. I can rebase the other one or just scratch it.

**Comment by rfourquet** [comment]
Thanks @KristofferC ! 
Someone would like to comment on this change before merging?

File: base/math.jl [hunk 1]
```diff
@@ -484,10 +484,12 @@ julia> hypot(a, a)
 1.4142135623730951e10
 
 julia> √(a^2 + a^2) # a^2 overflows
-ERROR: DomainError with -2914184810805067776:
-sqrt will only return a complex result if called with a complex argument. Try sqrt(complex(x)).
+ERROR: DomainError with -2.914184810805068e18:
```
Comment (tkelman) [inline base/math.jl]: why is this a float?

File: doc/src/manual/types.md [hunk 2]
```diff
@@ -1318,7 +1318,8 @@ for cases where you don't need a more elaborate hierarchy.
 ```jldoctest valtype
 julia> struct Val{x}
        end
-Base.@pure Val(x) = Val{x}()
+       Base.@pure Val(x) = Val{x}()
```
Comment (tkelman) [inline doc/src/manual/types.md]: this seems like it should have its own prompt line?

Comment (rfourquet) [inline doc/src/manual/types.md]: I agree yes, will update.

**Comment by rfourquet** [comment]
CI failure is a timeout.

**Pull request merged**